docs/ballooning.md

```diff
@@ -23,10 +23,14 @@ The device can be configured with the following options:
   allocate some memory which would make the guest enter an out-of-memory state,
   the kernel will take some pages from the balloon and give them to said
   process instead asking the OOM killer process to kill some processes to free
-  memory. Note that this applies to allocations from guest processes which would
-  make the system enter an OOM state. This does not apply to instances when the
-  kernel needs memory for its activities (i.e. constructing caches), or when the
-  user requests more memory than the amount available through an inflate.
+  memory. Note that this applies to physical page allocations in the kernel
+  which belong to guest processes. This does not apply to instances when the
+  kernel needs memory for its activities (i.e. constructing caches), when the
+  user requests more memory than the amount available through an inflate, or
+  when guest processes try to allocate large amounts of memory that are refused
+  by the guest memory manager (possible when the guest runs with
+  `vm.overcommit_memory=0` and the allocation does not pass the MM basic
+  checks).
```

Comment:
I would end this paragraph with smth like:
```
In conclusion, `deflate_on_oom` is also a best-effort mechanism and does not guarantee that no OOM can happen in the guest.
```

Reply:
This is partly true. If we say `deflate_on_oom` is best-effort, it would convey that its behavior is non-deterministic (that's how I would understand it at least), which is not the case. The behavior is clearly defined, it's just the conditions that are hard to define. I tried to capture this by mentioning `vm.memory_overcommit` and its behavior.

I could add something like:
```
Setting vm.memory_overcommit to 1 would make the MM approve all allocations, no matter how large, and using the memory mapped for those allocations will always deflate the balloon instead of making the guest enter an OOM state. Note: we do not recommend setting this to 1 because........
```
WDYT?

Reply:
Sounds good.